GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -0,0 +1,263 @@
+    // Remove it from the cache to avoid the font getting unloaded.
+    delete this._fontTobeUnloaded.name;
+  }
+};
+
+/**
+ * When an bitmapText object is removed, decrease the count of objects related to the font in the manager.
+ * When a font is not unused anymore, it goes in a temporary cache. The cache holds up to 10 fonts.
+ * If the cache reaches its maximum capacity, the oldest font is unloaded from memory.
+ * @param {string} name Slug of the bitmapFont
+ */
+gdjs.BitmapFontManager.prototype.removeFontUsed = function (name) {
+  if (!this._fontUsed[name]) {
+    // We tried to remove font that was never marked as used.
+    console.error(
+      'BitmapFont with name ' +
+        name +
+        ' was tried to be removed but was never marked as used.'
+    );
+    return;
+  }
+
+  this._fontUsed[name].objectsUsingTheFont--;
+
+  if (this._fontUsed[name].objectsUsingTheFont <= 0) {
+    delete this._fontUsed[name];
+
+    // Add the font name at the last position of the cache.
+    if (!this._fontTobeUnloaded.includes(name))
+      this._fontTobeUnloaded.push(name);
+
+    if (this._fontTobeUnloaded.length > 10) {
+      // Remove the first font (i.e: the oldest one)
+      const oldestUnloadedFont = this._fontTobeUnloaded.shift();
+      PIXI.BitmapFont.uninstall(oldestUnloadedFont);
+      console.log('Uninstall bitmapFont: ' + oldestUnloadedFont);
+    }
+  }
+};
+
+/**
+ * @param {string} bitmapFontResourceName
+ * @param {PIXI.Texture} texture
+ * @returns {PIXI.BitmapFont}
+ */
+gdjs.BitmapFontManager.prototype.getBitmapFontFromData = function (
+  bitmapFontResourceName,
+  texture
+) {
+  // Reuse an existing bitmapFont that is already in memory and already installed
+  for (let i = 0; i < this._loadedFontsData.length; ++i) {
+    let loadedFontData = this._loadedFontsData[i];
+    if (
+      loadedFontData.fontResourceFileName === bitmapFontResourceName &&
+      PIXI.BitmapFont.available[loadedFontData.fontName]
+    ) {
+      return PIXI.BitmapFont.available[loadedFontData.fontName];
+    }
+  }
+
+  let bitmapFontResource = this._resources.find(function (resource) {
+    return (
+      resource.kind === 'bitmapFont' && resource.name === bitmapFontResourceName
+    );
+  });
+
+  //A file was given, but the resource doesn't exist
+  if (bitmapFontResourceName !== '' && !bitmapFontResource) {
+    console.warn(
+      'Unable to find bitmap font for resource "' +
+        bitmapFontResourceName +
+        '" (or is not a bitmapFont file).'
+    );
+  }
+
+  // There is no bitmap font file or resource
+  if (!bitmapFontResource || bitmapFontResourceName === '') {
+    // If the default Arial bitmap font doesn't exist, generate it and use it.
+    if (!PIXI.BitmapFont.available[this.getDefaultSlugFontName()]) {
+      return this.getDefaultBitmapFont();
+    }
+
+    return PIXI.BitmapFont.available[this.getDefaultSlugFontName()];
```

Comment:
You should probably replace all of this by: `return getDefaultBitmapFont();`

That's the interest of keeping things in the function: you just call it whenever you need it, and it will generate+install the default font if necessary. As the caller, you don't care.

Reply:
I need to pass the font size to the Bitmap Text object when i update it [here](https://github.com/Bouh/GDevelop/blob/bitmap_font/Extensions/BitmapText/bitmaptextruntimeobject-pixi-renderer.js#L73-L78).

Reply:
Maybe I missed something, but I don't see the font size in these lines, so you could probably do:

```suggestion
    return this.getDefaultBitmapFont();
```

The default bitmap font is, well a default font, so there is no notion of font size?